Assignment 6

task 1

List all possible outcomes (sample space) for rolling a standard six-sided die and flipping a coin at the same time. Write your answer as a list of ordered pairs.

In [1]:
import itertools

die_faces = [1, 2, 3, 4, 5, 6]
coin_faces = ['H', 'T']

sample_space = list(itertools.product(die_faces, coin_faces))

print("Sample space (die, coin):")
print(sample_space)
print("\nTotal outcomes:", len(sample_space))


Sample space (die, coin):
[(1, 'H'), (1, 'T'), (2, 'H'), (2, 'T'), (3, 'H'), (3, 'T'), (4, 'H'), (4, 'T'), (5, 'H'), (5, 'T'), (6, 'H'), (6, 'T')]

Total outcomes: 12


**Explanation:** An experiment here is "roll a die and flip a coin together," and each ordered pair like (3, 'H') is one possible outcome — one specific die face paired with one specific coin face. The sample space is simply the set of all such pairs, and since the die has 6 faces and the coin has 2 sides, there are 6 × 2 = 12 equally likely outcomes in total. Any "event" of interest, such as "rolling an even number and getting heads," is just a subset of this sample space.

task 2

Suppose you have a deck of 10 cards numbered 1 to 10. Calculate the classical probability of drawing a card with an even number. Show your calculation steps.

In [2]:
import numpy as np

cards = np.arange(1, 11)
even_cards = cards[cards % 2 == 0]

favorable_outcomes = len(even_cards)
total_outcomes = len(cards)
probability_even = favorable_outcomes / total_outcomes

print("All cards:", cards)
print("Even-numbered cards:", even_cards)
print(f"P(even) = {favorable_outcomes} / {total_outcomes} = {probability_even}")


All cards: [ 1  2  3  4  5  6  7  8  9 10]
Even-numbered cards: [ 2  4  6  8 10]
P(even) = 5 / 10 = 0.5


**Explanation:** Classical probability is calculated as (number of favorable outcomes) divided by (total number of equally likely outcomes). Out of the 10 cards (1 through 10), exactly 5 are even: 2, 4, 6, 8, and 10. So the probability of drawing an even card is 5/10 = 0.5, meaning there's a 50% chance. This makes intuitive sense since exactly half of the numbers from 1 to 10 are even.

task 3

In a playlist of 8 songs on Spotify, 3 are by Arijit Singh. If you shuffle and play one random song, what is the probability that it is NOT by Arijit Singh? Show your working.

In [3]:
import numpy as np

total_songs = 8
arijit_songs = 3
other_songs = total_songs - arijit_songs

p_arijit = np.divide(arijit_songs, total_songs)
p_not_arijit = np.divide(other_songs, total_songs)
# equivalently: p_not_arijit = 1 - p_arijit

print("Total songs:", total_songs)
print("Arijit Singh songs:", arijit_songs)
print("Other songs:", other_songs)
print(f"P(Arijit Singh) = {arijit_songs}/{total_songs} = {p_arijit}")
print(f"P(NOT Arijit Singh) = {other_songs}/{total_songs} = {p_not_arijit}")
print("Check using complement rule: 1 - P(Arijit) =", np.subtract(1, p_arijit))


Total songs: 8
Arijit Singh songs: 3
Other songs: 5
P(Arijit Singh) = 3/8 = 0.375
P(NOT Arijit Singh) = 5/8 = 0.625
Check using complement rule: 1 - P(Arijit) = 0.625


**Explanation:** This uses the complement rule: P(not A) = 1 - P(A). Since 3 out of 8 songs are by Arijit Singh, P(Arijit) = 3/8 = 0.375. The remaining 5 songs are by other artists, so P(not Arijit) = 5/8 = 0.625. Both approaches — directly counting the 5 non-Arijit songs, or subtracting P(Arijit) from 1 — give the same answer, confirming there's a 62.5% chance the random song played is not by Arijit Singh.

task 4

A Zomato user places an order. The probability they order Pizza is 0.3, and the probability they order Burger is 0.4. If the probability they order both is 0.1, use the addition rule to find the probability the user orders Pizza or Burger.

**Hint:** P(A or B) = P(A) + P(B) - P(A and B)

In [4]:
import numpy as np

p_pizza = 0.3
p_burger = 0.4
p_both = 0.1

p_pizza_or_burger = np.add(p_pizza, p_burger) - p_both

print(f"P(Pizza) = {p_pizza}")
print(f"P(Burger) = {p_burger}")
print(f"P(Pizza and Burger) = {p_both}")
print(f"P(Pizza or Burger) = {p_pizza} + {p_burger} - {p_both} = {round(p_pizza_or_burger, 2)}")


P(Pizza) = 0.3
P(Burger) = 0.4
P(Pizza and Burger) = 0.1
P(Pizza or Burger) = 0.3 + 0.4 - 0.1 = 0.6


**Explanation:** The addition rule accounts for overlap between two events. If we simply added P(Pizza) + P(Burger) = 0.3 + 0.4 = 0.7, we would be double-counting the users who order both items. Subtracting P(Pizza and Burger) = 0.1 removes that double-count, giving P(Pizza or Burger) = 0.6. So there's a 60% chance a random Zomato user orders pizza, burger, or both in that order.

task 5

A Flipkart user is browsing electronics. The probability they add a phone to cart is 0.2. If you know they are interested in mobiles, the probability rises to 0.5. Briefly explain in your own words what this conditional probability means, using this scenario.

In [5]:
import numpy as np

p_add_to_cart = 0.2                 # P(Add phone to cart) - overall, unconditional
p_add_to_cart_given_mobile = 0.5    # P(Add phone to cart | Interested in mobiles)

increase_factor = np.divide(p_add_to_cart_given_mobile, p_add_to_cart)

print(f"P(Add to cart) = {p_add_to_cart}")
print(f"P(Add to cart | Interested in mobiles) = {p_add_to_cart_given_mobile}")
print(f"\nKnowing the user is interested in mobiles increases the probability by a factor of "
      f"{increase_factor}x")


P(Add to cart) = 0.2
P(Add to cart | Interested in mobiles) = 0.5

Knowing the user is interested in mobiles increases the probability by a factor of 2.5x


**Explanation:** Conditional probability measures how the chance of an event changes once we have extra information about another related event. Here, without knowing anything else about the user, there's only a 20% chance they add a phone to their cart. But once we learn they're specifically interested in mobiles (they're browsing that category, searching mobile brands, etc.), that same probability jumps to 50% — a 2.5x increase. This shows that "interest in mobiles" is strongly linked to the "add phone to cart" action, which is exactly the kind of insight Flipkart could use to target mobile-interested users with more relevant recommendations or offers.